src/main/scala/org/broadinstitute/dsde/agora/server/webservice/handlers/QueryHandler.scala

```diff
@@ -84,6 +90,22 @@ class QueryHandler(dataSource: PermissionsDataSource, implicit val ec: Execution
     }
   }
 
+  def queryPublic(requestContext: RequestContext, entity: AgoraEntity, descriptorType: ToolDescriptorType.DescriptorType): Future[PerRequestMessage] = {
+    val entityTypes = Seq(entity.entityType.getOrElse(throw new ValidationException("need an entity type")))
+    agoraBusiness.findSingle(entity, entityTypes, AccessControl.publicUser) map { foundEntity =>
+
+      descriptorType match {
+        case ToolDescriptorType.WDL =>
+          val result = ToolDescriptor(foundEntity.url.getOrElse(""),
```

Comment:
The urls returned by agora are known to be incorrect; this is GAWB-1741. I have chosen to continue to return incorrect urls here, like we do elsewhere. Once GAWB-1741 is fixed, these urls will magically be correct.

The other option is to always return the empty string for url, and once GAWB-1741 is fixed, we'd need to come back here to add the url back in. Reviewer: which is better?